### ISSUE TITLE
[Feature request] Having a removeGl*() function 

### ISSUE THREAD

**Issue by ColinFay** [r-spatial/leafgl#16]
`{leaflet}` allows to `leaflet::removeMarker()` (and other `remove*`) so that in a Shiny App you can trigger removal of elements from the server side. 

Would be nice to have a `removeGlPoints()` (and friends) to do the same thing on a map rendered with `{leafgl}`. 

I can dig into that if you think that's a good idea.

**Comment by tim-salabim** [comment]
Absolutely, I think it would be more than desirable to mimic leaflet behaviour as close as possible. Given that you're much more experienced when it comes to shiny, I'd very much welcome a PR if you don't mind.

**Comment by ColinFay** [comment]
Sure thing. I'll let you know!

**Comment by colinr23** [comment]
Hi great library - wondering if it is known to work with clearShapes() / leafletProxy() in shiny..or would this be the intended functionality of removeGl..? 
Cannot seem to get `leafletProxy("mymap") %>% clearShapes()` to work

**Comment by tim-salabim** [comment]
@ColinFay any update on this? And any comment on the question by @colinr23 ?

**Comment by colinr23** [comment]
follow up: `clearGroup(group='grp')` does work as expected, could not get `clearShapes()` to work

**Comment by ColinFay** [comment]
Sorry so far I hadn't had time to work on that :/ 

Will let you know

**Comment by ColinFay** [comment]
Hey folks, 

Here's my current hack to remove, hide or show things on a leafgl map in Shiny. 

``` r 
library(mapview)
library(leaflet)
library(leafgl)
library(sf)
library(shiny)

n = 1e6

df1 = data.frame(id = 1:n,
                 x = rnorm(n, 10, 3),
                 y = rnorm(n, 49, 1.8))

pts = st_as_sf(df1, coords = c("x", "y"), crs = 4326)

options(viewer = NULL) # view in browser

m = leaflet() %>%
  addProviderTiles(provider = providers$CartoDB.DarkMatter) %>%
  addGlPoints(data = pts, group = "pts") %>%
  addMouseCoordinates() %>%
  setView(lng = 10.5, lat = 49.5, zoom = 4) %>% 
  addLayersControl(overlayGroups = "pts")

## Shiny things ----

send_leafgl <- function(
  type = c("remove", "hide", "show", "toggle"), 
  id, 
  session = shiny::getDefaultReactiveDomain()
) {
  type <- match.arg(type)
  session$sendCustomMessage(
    sprintf("%s_leafgl", type), 
    id
  )
}

rm_lfgl <- purrr::partial(send_leafgl, type = "remove")
hide_lfgl <- purrr::partial(send_leafgl, type = "hide")
show_lfgl <- purrr::partial(send_leafgl, type = "show")
toggle_lfgl <- purrr::partial(send_leafgl, type = "toggle")

ui <- fluidPage(
  tags$script(
    "$( document ).ready(function() {
      Shiny.addCustomMessageHandler('remove_leafgl', function(id) {
        $('#' + id + ' canvas').remove();
      })
      Shiny.addCustomMessageHandler('hide_leafgl', function(id) {
        $('#' + id + ' canvas').hide();
      })
      Shiny.addCustomMessageHandler('show_leafgl', function(id) {
        $('#' + id + ' canvas').show();
      })
      Shiny.addCustomMessageHandler('toggle_leafgl', function(id) {
        $('#' + id + ' canvas').toggle();
      })
    });"
  ),
  leafglOutput("mymap"), 
  actionButton("remove", "remove"),
  actionButton("hide", "hide"),
  actionButton("show", "show"),
  actionButton("toggle", "toggle")
)

server <- function(input, output, session) {
  output$mymap <- renderLeaflet({
    m
  })
  observeEvent( input$remove , {
    rm_lfgl("mymap")
  })
  
  observeEvent( input$hide , {
    hide_lfgl("mymap")

  })
  
  observeEvent( input$show , {
    show_lfgl("mymap")
  })
  
  observeEvent( input$toggle , {
    toggle_lfgl("mymap")
  })
  
}

shinyApp(ui, server)
```

This isn't similar to what `{leaflet}` does though, but that seems like a nice feature to have.

**Comment by tim-salabim** [comment]
Nice! How could we integrate this into **leafgl**? And is the `purr::` dependency absolutely necessary?

**Comment by tim-salabim** [comment]
@alandipert I think this should be tackled before CRAN?

**Comment by tim-salabim** [comment]
This should be resolved with #33
